src/main/java/com/connexta/search/query/QueryManagerImpl.java

```diff
@@ -104,24 +105,26 @@ static Filter getFilter(final String queryString)
   }
 
   /**
-   * Creates a {@link List} of Product retrieve {@link URI}s from the {@code matchingIds} using the
-   * provided {@link #productRetrieveEndpoint}
+   * Creates a {@link List} of IRM retrieve {@link URI}s from the {@code matchingIds} using the
+   * provided {@link #datasetRetrieveEndpoint}
    *
    * @param matchingIds may be empty
-   * @return A {@link List} of product retrieve {@link URI}s
+   * @return A {@link List} of IRM retrieve {@link URI}s
    * @throws QueryException if unable to construct a retrieve URI
    */
-  private List<URI> getProductUris(final List<String> matchingIds) {
+  private List<URI> getIrmUris(final List<String> matchingIds) {
     return matchingIds.stream()
         .map(
             id -> {
               try {
-                return new URI(productRetrieveEndpoint + id);
+                final String irmRetrieveEndpoint =
+                    String.format("%s/%s/%s", datasetRetrieveEndpoint, id, IRM_PATH_SEGMENT);
```

Comment:
It is safer to use something like the [UriComponentBuilder](https://www.baeldung.com/spring-uricomponentsbuilder) to construct URIs than to use template strings (or string concatenation). Classes like UriComponentBuilder and Java's URI understand the domain of URIs. They help ensure that only valid URIs are constructed, and that characters and strings are properly escaped.